model/riscv_pte.sail

```diff
@@ -42,23 +42,32 @@ function to_pte_check(b : bool) -> PTE_Check =
  */
 function checkPTEPermission(ac : AccessType(ext_access_type), priv : Privilege, mxr : bool, do_sum : bool, p : PTE_Bits, ext : extPte, ext_ptw : ext_ptw) -> PTE_Check = {
   match (ac, priv) {
-    (Read(Data),      User)       => to_pte_check(p.U() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Write(Data),     User)       => to_pte_check(p.U() == 0b1 & p.W() == 0b1),
-    (ReadWrite(Data), User)       => to_pte_check(p.U() == 0b1 & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Execute(),       User)       => to_pte_check(p.U() == 0b1 & p.X() == 0b1),
+    (Read(Data),           User)       => to_pte_check(p.U() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
+    (Write(Data),          User)       => to_pte_check(p.U() == 0b1 & p.W() == 0b1),
+    (ReadWrite(Data,Data), User)       => to_pte_check(p.U() == 0b1 & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
+    (Execute(),            User)       => to_pte_check(p.U() == 0b1 & p.X() == 0b1),
 
-    (Read(Data),      Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Write(Data),     Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & p.W() == 0b1),
-    (ReadWrite(Data), Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
-    (Execute(),       Supervisor) => to_pte_check(p.U() == 0b0 & p.X() == 0b1),
+    (Read(Data),           Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
+    (Write(Data),          Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & p.W() == 0b1),
+    (ReadWrite(Data,Data), Supervisor) => to_pte_check((p.U() == 0b0 | do_sum) & p.W() == 0b1 & (p.R() == 0b1 | (p.X() == 0b1 & mxr))),
+    (Execute(),            Supervisor) => to_pte_check(p.U() == 0b0 & p.X() == 0b1),
 
-    (_,               Machine)    => internal_error("m-mode mem perm check")
+    (_,                    Machine)    => internal_error("m-mode mem perm check")
   }
 }
 
 function update_PTE_Bits(p : PTE_Bits, a : AccessType(ext_access_type), ext : extPte) -> option((PTE_Bits, extPte)) = {
-  let update_d = (a == Write(Data) | a == ReadWrite(Data)) & p.D() == 0b0; // dirty-bit
-  let update_a = p.A() == 0b0;                                             // accessed-bit
+  // dirty bit
+  let update_d : bool = match a {
+                          Read(Data)      => false,
+                          Execute()       => false,
+                          Write(Data)     => true,
+                          ReadWrite(Data) => true
+                        };
+  let update_d = update_d & p.D() == 0b0;
+
+  // accessed bit
+  let update_a = p.A() == 0b0;
```

Comment:
This seems unnecessary, and also gets the rarity of ReadWrite data still wrong. I'd just add `, Data` to the existing code (and at some point I'd like to refactor this to match on `Read(_)` etc and call an `ext_update_PTE_Bits` so we can avoid duplicating the code in sail-cheri-riscv, with suitable changes to `checkPTEPermission` etc).